Sources/SwiftDocC/Semantics/Metadata/Availability.swift

```diff
@@ -38,19 +46,42 @@ extension Metadata {
     public final class Availability: Semantic, AutomaticDirectiveConvertible {
         static public let directiveName: String = "Available"
 
-        public enum Platform: String, RawRepresentable, CaseIterable, DirectiveArgumentValueConvertible {
+        public enum Platform: RawRepresentable, Hashable, DirectiveArgumentValueConvertible {
             // FIXME: re-add `case any = "*"` when `isBeta` and `isDeprecated` are implemented
             // cf. https://github.com/apple/swift-docc/issues/441
             case macOS, iOS, watchOS, tvOS
 
+            case other(String)
+
+            static var defaultCases: [Platform] = [.macOS, .iOS, .watchOS, .tvOS]
+
             public init?(rawValue: String) {
-                for platform in Self.allCases {
+                for platform in Self.defaultCases {
                     if platform.rawValue.lowercased() == rawValue.lowercased() {
                         self = platform
                         return
                     }
                 }
-                return nil
+                if rawValue == "*" {
```

Comment:
With this implementation as well, we can mimic Deprecated behaviors as well? it would be so confusing to have @Available() with Deprecated passed to it.

Reply:
Do you mean allowing `@Available(isDeprecated: true)` once that is implemented? That will need to be handled as part of that implementation.

If you mean allowing `@Available(MyPackage, isDeprecated, true)`, that should come for free without any extra work. I could add a test for that if you would like.

Reply:
I meant the first part, i.e  @Available(isDeprecated: true). sorry my response was never posted due to my internet connectivity.

Reply:
Right, deprecation status isn't available yet; see https://github.com/apple/swift-docc/issues/441 for information.